path: Normalize paths

johannes-wolf opened the pull request.

fenjalien commented on src/path-util.typ:
```diff
@@ -176,6 +176,41 @@
+
+      if kind == "line" and head.at(0) == kind {
+        // Merge consecutive line segments
+        new.last() += pts.slice(1)
```
Why are `pts` being sliced? Doesn't that drop the first point of the line?

fenjalien commented on src/path-util.typ:
```diff
@@ -176,6 +176,41 @@
+      if kind == "line" and head.at(0) == kind {
+        // Merge consecutive line segments
+        new.last() += pts.slice(1)
+      } else if segment-start(s) != segment-end(head) {
+        // Push a new line or line point if the current segment
+        // does not start where the previous segment ended
+        if head.at(0) == "line" {
+          new.last().push(segment-start(s))
```
Why not use `pts.first()` instead of `segment-start`?

fenjalien commented:
Oh and how could we go about adding a test for this? Or would it be covered by a merge path test?

johannes-wolf commented:
> Oh and how could we go about adding a test for this? Or would it be covered by a merge path test?

Fixed. I've added a test case to the `merge-path` test.
